public/app/plugins/panel/annolist/AnnoListPanel.tsx

```diff
@@ -113,7 +113,9 @@ export class AnnoListPanel extends PureComponent<Props, State> {
     }
 
     if (options.tags && options.tags.length) {
-      params.tags = options.tags;
+      params.tags = options.tags.map((tag) =>
+        this.props.replaceVariables(tag) ? this.props.replaceVariables(tag) : tag
```

Comment:
I guess you meant to do this?:
```suggestion
        this.props.replaceVariables ? this.props.replaceVariables(tag) : tag
```

Reply:
I tested it, and it seems that the `replaceVariables` function is always defined, but returns `undefined` if the string does not contains a variable.

I can change it to the following, if needed:

```js
params.tags = options.tags.map((tag) => {
  const newTag = this.props.replaceVariables(tag);
  return newTag ? newTag : tag;
});
```

Reply:
@torkelo is this true? Does the replaceVariables return `undefined` if the string doesn't contain a variable? This doesn't sound right to me.

Reply:
@hugohaggmark 
yes, that sounds wrong, it returns the original string  (or should)

Reply:
Okay, my bad. After further investigations, it seems that `replaceVariables` is always defined, but returns `undefined` when running `yarn test` but returns the expected result when running with `yarn start`. So there is still a problem here, but it's a little out of my scope, sorry 😞

Reply:
I'll see what I can do to fix the tests

Reply:
@francoisdtm if you change this line https://github.com/francoisdtm/grafana/blob/0c554938db2005e6d8949f77486c8e569f120b70/public/app/plugins/panel/annolist/AnnoListPanel.test.tsx#L79

to become

```ts
replaceVariables: (str: string) => str,
```

Reply:
Perfect, it works as expected. Thanks a lot!